core/domain/topic_jobs_one_off_test.py

```diff
@@ -340,3 +340,135 @@ def test_job_skips_deleted_topic(self):
+            language_code='en',
+            canonical_story_references=[],
+            additional_story_references=[],
+            uncategorized_skill_ids=[],
+            subtopic_schema_version=feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION,
+            story_reference_schema_version=(
+                feconf.CURRENT_STORY_REFERENCE_SCHEMA_VERSION),
+            next_subtopic_id=1,
+            subtopics=[]
+        )
+        commit_message = (
+            'New topic created with name \'Topic name\'.')
+
+        topic_models.TopicRightsModel(
+            id=self.TOPIC_ID,
+            manager_ids=[self.albert_id],
+            topic_is_published=True
+        ).commit(
+            self.albert_id, 'Created new topic rights',
+            [{'cmd': topic_domain.CMD_CREATE_NEW}])
+
+        topic_model.commit(
+            self.albert_id, commit_message, [{
+                'cmd': topic_domain.CMD_CREATE_NEW,
+                'name': 'Topic name'
+            }])
+
+        # The topic summary model isn't created yet.
+        topic_summary_model = (
+            topic_models.TopicSummaryModel.get(self.TOPIC_ID, strict=False))
+        self.assertIsNone(topic_summary_model)
+
+        # Start migration job.
+        job_id = (
+            topic_jobs_one_off.RegenerateTopicSummaryOneOffJob.create_new())
+        topic_jobs_one_off.RegenerateTopicSummaryOneOffJob.enqueue(job_id)
+        self.process_and_flush_pending_tasks()
+
+        # Verify the topic summary is created correctly.
+        topic_summary_model = (
+            topic_models.TopicSummaryModel.get(self.TOPIC_ID, strict=False))
+        self.assertEqual(
+            topic_summary_model.thumbnail_filename, 'topic.svg')
+        self.assertEqual(
+            topic_summary_model.thumbnail_bg_color, '#C6DCDA')
+
+        output = topic_jobs_one_off.RegenerateTopicSummaryOneOffJob.get_output(
+            job_id)
+        expected = [[u'topic_processed',
+                     [u'Successfully processed 1 topics.']]]
+        self.assertEqual(expected, [ast.literal_eval(x) for x in output])
+
+    def test_regeneration_job_skips_invalid_topic(self):
+
+        def _mock_get_topic_by_id(unused_topic_id):
+            """Mocks get_topic_by_id()."""
+            return 'invalid_topic'
+
+        topic = topic_domain.Topic.create_default_topic(
+            self.TOPIC_ID, 'A title', 'Abbrev title', 'description')
+        topic_services.save_new_topic(self.albert_id, topic)
+
+        get_topic_by_id_swap = self.swap(
+            topic_fetchers, 'get_topic_by_id', _mock_get_topic_by_id)
+
+        with get_topic_by_id_swap:
+            job_id = (
+                topic_jobs_one_off.RegenerateTopicSummaryOneOffJob.create_new())
+            topic_jobs_one_off.RegenerateTopicSummaryOneOffJob.enqueue(job_id)
+            self.process_and_flush_pending_tasks()
+
+        output = topic_jobs_one_off.RegenerateTopicSummaryOneOffJob.get_output(
+            job_id)
+
+        for x in output:
```

Comment:
Perhaps give a more semantic name for x.

Reply:
Done